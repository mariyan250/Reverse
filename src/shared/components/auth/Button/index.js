// React and style
import React from 'react';
import './index.scss';

function Button(props) {
  return (
    <button className='auth-btn' {...props}>
      {props.text}
    </button>
  );
}

export default Button;
