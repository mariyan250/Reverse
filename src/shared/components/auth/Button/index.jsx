// React and style
import React from 'react';
import styles from './index.module.scss';

function Button({ text, onClick, className }) {
  return (
    <section className={`${styles.button} ${className}`}>
      <button className={`app-bg-primary border-0 app-text-white w-100 p-2`} onClick={onClick}>
        {text}
      </button>
    </section>
  );
}

export default Button;
