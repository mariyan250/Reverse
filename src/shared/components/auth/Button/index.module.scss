@import '../../../../index.scss';

.button {
  font-size: to-rem(18px);
  animation: appear 1 $timing-slow;

  &:hover {
    background: $primary-hover;
  }
}
