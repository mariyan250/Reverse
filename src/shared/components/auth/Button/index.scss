@import '../../../../index.scss';

// rethink, rewrite everything.
.auth-btn {
  background: $background-primary;
  padding: $spacing-xs;
  font-size: 18px;
  border: none;
  outline: none;
  color: $text-white;
  margin-top: $spacing-s;
  animation: appear 1 0.5s;

  &:hover {
    background: $background-primary-hover;
    cursor: pointer;
  }
}
