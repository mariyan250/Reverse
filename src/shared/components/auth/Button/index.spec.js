import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import Button from './index';

describe('Button', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { asFragment } = renderWithRouter(<Button />, initialState);

      expect(asFragment(<Button />)).toMatchSnapshot();
    });
  });
});
