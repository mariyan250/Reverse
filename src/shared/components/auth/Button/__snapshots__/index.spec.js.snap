// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the component 1`] = `
<div>
  <button
    class="button app-bg-primary border-0 app-text-white w-100 p-2"
  />
</div>
`;
